crates/core/src/db/queries/metadatas.rs

```diff
@@ -503,79 +509,124 @@ pub fn collection_nfts(conn: &Connection, options: CollectionNftOptions) -> Resu
+                (Metadatas::Table, Metadatas::UpdateAuthorityAddress),
+                (Metadatas::Table, Metadatas::MintAddress),
+                (Metadatas::Table, Metadatas::PrimarySaleHappened),
+                (Metadatas::Table, Metadatas::Uri),
+                (Metadatas::Table, Metadatas::Slot),
+            ])
+            .columns(vec![
+                (MetadataJsons::Table, MetadataJsons::Description),
+                (MetadataJsons::Table, MetadataJsons::Image),
+                (MetadataJsons::Table, MetadataJsons::AnimationUrl),
+                (MetadataJsons::Table, MetadataJsons::ExternalUrl),
+                (MetadataJsons::Table, MetadataJsons::Category),
+                (MetadataJsons::Table, MetadataJsons::Model),
+            ])
+            .columns(vec![(
                 CurrentMetadataOwners::Table,
-                CurrentMetadataOwners::MintAddress,
-            ),
-        )
-        .left_join(
-            Listings::Table,
-            Condition::all()
-                .add(
-                    Expr::tbl(Listings::Table, Listings::Metadata)
-                        .equals(Metadatas::Table, Metadatas::Address),
-                )
-                .add(Expr::tbl(Listings::Table, Listings::Seller).equals(
+                CurrentMetadataOwners::TokenAccountAddress,
+            )])
+            .from(MetadataJsons::Table)
+            .inner_join(
+                Metadatas::Table,
+                Expr::tbl(MetadataJsons::Table, MetadataJsons::MetadataAddress)
+                    .equals(Metadatas::Table, Metadatas::Address),
+            )
+            .inner_join(
+                CurrentMetadataOwners::Table,
+                Expr::tbl(Metadatas::Table, Metadatas::MintAddress).equals(
                     CurrentMetadataOwners::Table,
-                    CurrentMetadataOwners::OwnerAddress,
-                )),
-        )
-        .union(UnionType::All, Query::select()
-            .from(MeCollections::Table)
-            .expr_as(Expr::col(MeCollections::Id), Alias::new("address"))
-            .expr_as(Expr::col(MeCollections::Name), Alias::new("name"))
+                    CurrentMetadataOwners::MintAddress,
+                ),
+            )
+            .left_join(
+                Listings::Table,
+                Condition::all()
+                    .add(
+                        Expr::tbl(Listings::Table, Listings::Metadata)
+                            .equals(Metadatas::Table, Metadatas::Address),
+                    )
+                    .add(Expr::tbl(Listings::Table, Listings::Seller).equals(
+                        CurrentMetadataOwners::Table,
+                        CurrentMetadataOwners::OwnerAddress,
+                    )),
+            )
+            .and_where(Expr::col(Metadatas::BurnedAt).is_null())
+            .limit(limit)
+            .offset(offset)
+            .order_by((Listings::Table, Listings::Price), Order::Asc)
```

Comment:
The limit and offset will be the same for both queries consider assigning the base query to a variable then adding limit offset. 

Also order_by isn't always price. That will need some logic to add the order by based on the arguments to the function